engine/storage/image/src/org/apache/cloudstack/storage/image/TemplateServiceImpl.java

```diff
@@ -638,7 +638,9 @@ protected Void createTemplateAsyncCallBack(AsyncCallbackDispatcher<TemplateServi
     }
 
     private Map<String, TemplateProp> listTemplate(DataStore ssStore) {
-        ListTemplateCommand cmd = new ListTemplateCommand(ssStore.getTO(), imageStoreDetailsUtil.getNfsVersion(ssStore.getId()));
+        String nfsVersionParam = imageStoreDetailsUtil.getNfsVersion(ssStore.getId());
+        Integer nfsVersion = (nfsVersionParam != null ? Integer.valueOf(nfsVersionParam) : null);
```

Comment:
Please move the nfs version conversion from string to int to a separate method, saw the same pattern at multiple places.